### PR TITLE
Package Refactoring - Split Console Sink to new project

### PR DESCRIPTION
**Pull request by merbla**
I have a number of these sink ports on the go at once.  Wanted to get a basic template going relating to #593 so the others are rinse/repeat.

@nblumhardt @khellang what else is missing??

### CODE REVIEW

**Comment by khellang** [comment]
Looks like the build is falling because of a missing reference to [System.Dynamic.Runtime](https://www.nuget.org/packages/System.Dynamic.Runtime/4.0.11-beta-23516)...

File: src/Serilog.Sinks.Console/project.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    "net40": {
+      "compilationOptions": {
+        "keyFile": "../../assets/Serilog.snk"
+      },
+      "frameworkAssemblies": {
+        "System.Configuration": ""
+      },
+      "dependencies": {
+        "Microsoft.Bcl.Async": "1.0.168",
+        "Serilog": "2.0.0-beta-*"
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: This should be moved to the top-level `"dependencies"` node, outside the specific targets.

File: src/Serilog.Sinks.Console/project.json [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        "keyFile": "../../assets/Serilog.snk"
+      },
+      "frameworkAssemblies": {
+        "System.Configuration": ""
+      },
+      "dependencies": {
+        "Serilog": "2.0.0-beta-*"
+      }
+    },
+    "dotnet5.4": {
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: Is `dotnet5.4` the minimum we can support for this sink? I guess it depends on the `System.Console` package...

File: src/Serilog.Sinks.Console/project.json [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      "dependencies": {
+        "Microsoft.CSharp": "4.0.1-beta-23516",
+        "Serilog": "2.0.0-beta-*",
+        "System.Collections.Concurrent": "4.0.11-beta-23516",
+        "System.Console": "4.0.0-beta-23516",
+        "System.Diagnostics.Process": "4.1.0-beta-23516",
+        "System.Diagnostics.TraceSource": "4.0.0-beta-23516",
+        "System.IO": "4.0.11-beta-23516",
+        "System.IO.FileSystem": "4.0.1-beta-23516",
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: I'm not sure we need this? The dependencies should probably be shaved down for this package?

File: src/Serilog/LoggerConfigurationFullNetFxExtensions.cs [hunk 4]
```diff
@@ -322,7 +298,7 @@ public static class LoggerConfigurationFullNetFxExtensions
             {
                 if (settingPrefix.Contains(":")) throw new ArgumentException("Custom setting prefixes cannot contain the colon (:) character.");
                 if (settingPrefix == "serilog") throw new ArgumentException("The value \"serilog\" is not a permitted setting prefix. To use the default, do not specify a custom prefix at all.");
-                if (string.IsNullOrWhiteSpace(settingPrefix)) throw new ArgumentException("To use the default setting prefix, do not supply the settingPrefix parameter, instead pass the default null.");
+                if (String.IsNullOrWhiteSpace(settingPrefix)) throw new ArgumentException("To use the default setting prefix, do not supply the settingPrefix parameter, instead pass the default null.");
```
Comment (khellang) [inline src/Serilog/LoggerConfigurationFullNetFxExtensions.cs]: What's the reason for this change?

File: test/Serilog.Tests/project.json [hunk 5]
```diff
@@ -10,7 +10,8 @@
         "Serilog": { "target": "project" },
         "xunit": "2.1.0",
         "xunit.runner.visualstudio": "2.1.0",
-        "xunit.runner.dnx": "2.1.0-rc1-build204"
+      "xunit.runner.dnx": "2.1.0-rc1-build204",
```
Comment (khellang) [inline test/Serilog.Tests/project.json]: Nit: indentation

File: src/Serilog.Sinks.Console/project.json (hunk 1 above)
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: Also, it's probably better (for now) to change this to `{ "target": "project" }` so it doesn't pick up the NuGet.org package instead.

File: src/Serilog.Sinks.Console/project.json [hunk 6]
```diff
@@ -0,0 +1,60 @@
+{
+  "version": "2.0.0-beta-*",
+  "description": "The console sink for Seriolog - Simple .NET logging with fully-structured events",
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: Typo: Seriolog :wink:

File: src/Serilog.Sinks.Console/LoggingExtentsions.cs [hunk 7]
```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using Serilog.Configuration;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Display; 
+
+
```
Comment (khellang) [inline src/Serilog.Sinks.Console/LoggingExtentsions.cs]: Nit: double blank line

File: test/Serilog.Tests/project.json [hunk 8]
```diff
@@ -10,7 +10,8 @@
...
+      "Serilog.Sinks.Console": "2.0.0-beta-*"
```
Comment (khellang) [inline test/Serilog.Tests/project.json]: This should also be `{ "target": "project" }` :smile:

**Comment by khellang** [comment]
Nice work! :sparkles: Left some comments. I think it's important that we establish a baseline, as you say, before we start splitting out the rest :smile:

File: src/Serilog.Sinks.Console/project.json (hunk 2 above)
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: Looks like `System.Console` is `dotnet5.1`. That should be our aim for this package.

File: src/Serilog.Sinks.Console/project.json [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+        "keyFile": "../../assets/Serilog.snk"
+      },
+      "frameworkAssemblies": {
+        "System.Configuration": ""
+      },
+      "dependencies": {
+        "Serilog": "2.0.0-beta-*"
+      }
+    },
+    "dnx451": {
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: This target should :fire: DNX is going away and all MS' packages have removed the target. `net45` should work for the same platforms.

**Comment by merbla** [comment]
Sweet thanks @khellang.  Threw this together real quick.  Build.ps1 runs fine locally so have to dig a little deeper.

File: src/Serilog/LoggerConfigurationFullNetFxExtensions.cs (hunk 4 above)
Comment (merbla) [inline src/Serilog/LoggerConfigurationFullNetFxExtensions.cs]: :fire: done

File: src/Serilog.Sinks.Console/project.json (hunk 1 above)
Comment (merbla) [inline src/Serilog.Sinks.Console/project.json]: done

File: src/Serilog.Sinks.Console/project.json (hunk 2 above)
Comment (merbla) [inline src/Serilog.Sinks.Console/project.json]: done :fire: `dnx451`

File: src/Serilog.Sinks.Console/project.json (hunk 3 above)
Comment (merbla) [inline src/Serilog.Sinks.Console/project.json]: :scissors:

File: test/Serilog.Tests/project.json (hunk 5 above)
Comment (merbla) [inline test/Serilog.Tests/project.json]: Damn u :vs:

File: src/Serilog.Sinks.Console/LoggingExtentsions.cs (hunk 7 above)
Comment (merbla) [inline src/Serilog.Sinks.Console/LoggingExtentsions.cs]: :fire:

File: test/Serilog.Tests/project.json (hunk 8 above)
Comment (merbla) [inline test/Serilog.Tests/project.json]: I will never let VS author a `project.json` again

File: src/Serilog.Sinks.Console/LoggingExtentsions.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Serilog.Configuration;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Display; 
+
+namespace Serilog.Sinks.SystemConsole
+{
+    public static class LoggingExtentsions
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/LoggingExtentsions.cs]: I think we use `*LoggerConfigurationExtensions` in most other sinks - so `ConsoleLoggerConfigurationExtensions` here?

File: src/Serilog.Sinks.Console/project.json [hunk 11]
```diff
@@ -0,0 +1,38 @@
+{
+  "version": "2.0.0-beta-*",
+  "description": "The console sink for Serilog - Simple .NET logging with fully-structured events",
+  "authors": [ "Serilog Contributors" ],
+  "tags": [ "serilog", "logging", "semantic", "structured", "console" ],
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/project.json]: Perhaps just `"serilog", "console"`?

**Comment by nblumhardt** [comment]
Couple of nitpick comments, but looks good to me! So good to see 2.0 finally coming together - fingers crossed the rest of the world stays stable around us, and we'll have this out for RTM! :)

File: src/Serilog.Sinks.Console/project.json [hunk 12]
```diff
@@ -0,0 +1,38 @@
...
+      },
+      "dependencies": {
+        "Microsoft.Bcl.Async": "1.0.168"
+      }
+    },  
+    "net45": {
+      "compilationOptions": {
+        "keyFile": "../../assets/Serilog.snk"
+      },
+      "frameworkAssemblies": {
```
Comment (khellang) [inline src/Serilog.Sinks.Console/project.json]: :fire: these? Just noise to my :eyes:

**Comment by nblumhardt** [comment]
Looks great, ready to merge but will double check first that AppVeyor is going to publish the new package for us. Should have this in tonight.

File: src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs [hunk 13]
```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using Serilog.Configuration;
+using Serilog.Core;
+using Serilog.Events;
+using Serilog.Formatting.Display; 
+
+namespace Serilog.Sinks.SystemConsole
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs]: Namespace is wrong here, should be `Serilog`

File: src/Serilog.Sinks.Console/project.json [hunk 14]
```diff
@@ -0,0 +1,24 @@
+{
+  "version": "2.0.0-beta-*",
+  "description": "The console sink for Serilog - Simple .NET logging with fully-structured events",
+  "authors": [ "Serilog Contributors" ],
+  "tags": [ "serilog", "console" ],
+  "projectUrl": "http://serilog.net",
+  "licenseUrl": "http://www.apache.org/licenses/LICENSE-2.0",
+  "iconUrl": "http://serilog.net/images/serilog-nuget.png",
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/project.json]: Needs http://serilog.net/images/serilog-sink-nuget.png

File: src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs [hunk 15]
```diff
@@ -0,0 +1,38 @@
...
+{
+    public static class ConsoleLoggerConfigurationExtensions
+    {
+        const string DefaultConsoleOutputTemplate = "{Timestamp:yyyy-MM-dd HH:mm:ss} [{Level}] {Message}{NewLine}{Exception}";
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs]: Should this be removed from the original location now it has moved here?

File: src/Serilog.Sinks.Console/Serilog.Sinks.Console.xproj [hunk 16]
```diff
@@ -0,0 +1,20 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
```
Comment (nblumhardt) [inline src/Serilog.Sinks.Console/Serilog.Sinks.Console.xproj]: Referring to the renamed file above, which I can't comment on directly :-) The folder structure of the project should start at the root namespace Serilog, so the ConsoleSink.cs file belongs in a `Sinks\Console` subfolder.

**Comment by nblumhardt** [comment]
@merbla actually, off the back of some of the things you spotted in the other PR, a couple more comments :-) Cheers!

File: src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs (hunk 15 above)
Comment (merbla) [inline src/Serilog.Sinks.Console/ConsoleLoggerConfigurationExtensions.cs]: This was still used by ColoredConsole.  Next few sinks I will batch together.

**Comment by merbla** [comment]
@nblumhardt show me the squirrel

**Comment by khellang** [comment]
:shipit:

**Comment by nblumhardt** [comment]
:ship: :-)

**Comment by nblumhardt** [comment]
https://www.nuget.org/packages/serilog.sinks.console
